frontend/src/app/projects/project-detail/model-overview/model-detail/git-model.service.ts

```diff
@@ -86,4 +82,37 @@ export class GitModelService {
         })
       );
   }
+
+  addGitSource(
+    project_slug: string,
+    model_slug: string,
+    source: CreateGitModel
+  ): Observable<GetGitModel> {
+    return this.http.post<GetGitModel>(
+      environment.backend_url +
+        `/projects/${project_slug}/models/${model_slug}/modelsources/git`,
+      source
+    );
+  }
+}
+
+export interface BaseGitModel {
+  path: string;
+  revision: string;
+  entrypoint: string;
+  username: string;
+}
+
+export interface CreateGitModel extends BaseGitModel {
+  password?: string;
+}
+
+export interface PatchGitModel extends CreateGitModel {
+  primary: boolean;
+}
+
+export interface GetGitModel extends BaseGitModel {
+  id: number;
+  primary: boolean;
+  password: boolean;
 }
```

Comment:
It’s semantically more correct to use types instead of interfaces I think

Reply:
Done.